content/docs/going-deeper/tasks.md

```diff
@@ -151,7 +151,14 @@ call park, if an unpark happens it'll store `true` to `ready` (as we saw above
 in `Unpark for ThreadUnpark`). If this hasn't happened, though, then we block
 the current thread. The current thread will then get reawoken once the future's
 task is unparked, through the `ThreadUnpark` instance associated with this
-future.
+future. This must be done from another thread that is not parked. Some futures
+need a `Core` to be running in some thread to proceed, or some other trigger
+they registered into. Obviously a different one than which we block with `wait`.
```

Comment:
Yes, tasks are parked. However, if I follow the code correctly, in the case of the `wait` executor in this chapter, parking the task means parking the underlying thread. Citing the paragraph just above this modified one:

| Recall that we're just blocking the current thread until the future is
| finished, so we're just going to call [`thread::park`] here from the standard
| library. To get a better idea, let's take a look at the `ThreadUnpark::park`
| function we're calling:

And the code snippet also speaks quite clearly (also just from above of the paragraph I modify):

```rust,ignore
fn park(&self) {
    if !self.ready.swap(false, Ordering::SeqCst) {
        thread::park();
    }
}
```

Indeed, in the case of different executor (the `Core`, for example), parking the task doesn't mean parking the thread and the task can be unparked from the same thread. But this modification is about the [`wait` executor](https://tokio.rs/docs/going-deeper/tasks/#exploring-wait).

The problem I'm trying to solve here is, when I first read this part of docs and remembered the example with prime computation and timeout, I wondered how on Earth can this task ever continue, when I park the underlying thread ‒ and the answer to that is its `unpark` got called from another thread (either one with the computation, or one with the timeout). My confusion probably stemmed from the idea that I'm doing all this futures dancing because I want to *avoid* having other threads around, so why would I want to park my (only) thread ‒ but that kind of thinking is relevant to the `Core` executor, not here.